flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/hive/HivePropertiesConverter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.flink.connector.hive;
+
+import static com.datastrato.gravitino.catalog.hive.HiveCatalogPropertiesMeta.METASTORE_URIS;
+
+import com.datastrato.gravitino.flink.connector.PropertiesConverter;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import java.util.Map;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.catalog.CommonCatalogOptions;
+import org.apache.flink.table.catalog.hive.HiveCatalog;
+import org.apache.flink.table.catalog.hive.factories.HiveCatalogFactoryOptions;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+public class HivePropertiesConverter implements PropertiesConverter {
```

Comment:
1. when creating catalog, `toGravitinoCatalogProperties` load configurations from `HIVE_CONF_DIR` and saving all configurations with `flink.bypass` to Gravitino server, YES?
2. when loading catalogs, `toFlinkCatalogProperties` get all configurations with `flink.bypass` prefix we call it `GravitinoProperties`.  How does flink handle GravitinoProperties?  does flink need to read configuration from `HIVE_CONF_DIR` again?   Is `GravitinoProperties` equals to Context.getOptions in `GravitinoHiveCatalogFactory`?

Reply:
@FANNG1 
1. It will not save the configuration that loads from the `hive-conf-dir`,  it only saves the `hive-conf-dir` to the Gravitino. The configuration in 'CatalogDescription` must be some of the `requiredOptions` in `GravitinoHiveCatalogFactory`. 
2. Gravitino will pass the `hive-conf-dir` and other Hadoop configurations to the GravitinoHiveCatalogFactory. Then the Flink will create the HiveConf and the HiveCatalog in GravitinoHiveCatalog. The Hadoop configuration saved in Gravitino has higher priority， it will overwrite the configuration that loaded from `hive-conf-dir`. 

>  Is GravitinoProperties equals to Context.getOptions in GravitinoHiveCatalogFactory?

Yes. The `GravitinoProperties` is equal to the  `CatalogDescriptor.configuration`. And the `CatalogDescriptor.configuration` is saved in Gravitino. When creating the `GravitinoHiveCatalog`, it will load the configuration from the Gravitino, and remove the prefix `flink.bypass`, and then pass to the Context 

```
private Catalog initCatalog(String catalogName, CatalogDescriptor catalogDescriptor) {
        return FactoryUtil.createCatalog(
                catalogName,
                catalogDescriptor.getConfiguration().toMap(),
                catalogStoreHolder.config(),
                catalogStoreHolder.classLoader());
    }


public static Catalog createCatalog(
            String catalogName,
            Map<String, String> options,
            ReadableConfig configuration,
            ClassLoader classLoader) {
        // Use the legacy mechanism first for compatibility
        try {
            final CatalogFactory legacyFactory =
                    TableFactoryService.find(CatalogFactory.class, options, classLoader);
            return legacyFactory.createCatalog(catalogName, options);
        } catch (NoMatchingTableFactoryException e) {
            // No matching legacy factory found, try using the new stack

            final DefaultCatalogContext discoveryContext =
                    new DefaultCatalogContext(catalogName, options, configuration, classLoader);
            try {
                final CatalogFactory factory = getCatalogFactory(discoveryContext);

                // The type option is only used for discovery, we don't actually want to forward it
                // to the catalog factory itself.
                final Map<String, String> factoryOptions =
                        options.entrySet().stream()
                                .filter(
                                        entry ->
                                                !CommonCatalogOptions.CATALOG_TYPE
                                                        .key()
                                                        .equals(entry.getKey()))
                                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
                final DefaultCatalogContext context =
                        new DefaultCatalogContext(
                                catalogName, factoryOptions, configuration, classLoader);
                return factory.createCatalog(context);   // here is the options. come from the catalogDescription from the Gravitino.
            } 

```

Reply:
got it, thanks